src/Components/Login/Login.tsx

```diff
@@ -19,56 +13,61 @@ const ALL_USERS = gql `
   }
 `
 
-const ALL_BOOKS = gql `
-  query AllBooks {
-    books {
-      id
-      imageUrl
-    }
-  }
-`
-
 const Login = () => {
   const [email, setEmail] = useState('')
   const [username, setUsername] = useState('')
   const [userLocation, setUserLocation] = useState('')
   const [activeAccount, setAccount] = useState(true)
-  const [allUsers, setUsers] = useState([{userName: "Josh", emailAddress: "test", location: "Denver"}])
+  const [loginError, setLoginError] = useState(false)
+  const [errorMessage, setErrorMessage] = useState('')
+  const [allUsers, setUsers] = useState([{emailAddress: "joshua@gmail.com", userName: "Joshua", location: "Colorado"}])
```

Comment:
For all of the `useState` hooks above, the type is pretty easily inferred. It's a string, or a boolean.

For more complex types, like your user object, I'd recommend having an actual `type` or `interface` somewhere that you can use. This will give you the full power of Typescript and allow any and all devs working in this code to know what the shape of this data is supposed to be, thus preventing future goofs.

It looks like you have a user interface at the top of this file, but this PR removed it. I'm curious as to why, because you could be declaring your hook as:
```TS
const [allUsers, setUsers] = useState<User[]>([]);
```

This example brings up another question. Why are you defaulting your allUsers state with a hardcoded user?

Reply:
> This example brings up another question. Why are you defaulting your allUsers state with a hardcoded user?

Oooooh, I see. This is your source of truth for what emails are considered valid. I'm assuming this will be replaced by the results of the ALL_USERS query.

In this case, this probably doesn't need to be in state. It could be a ref, but let's not worry about that now and you guys continue developing as you had planned. Sorry!

Reply:
Thank you for all that feed back and yes the ALL_USERS query will be replacing the allUsers state.

Reply:
Updated this to include a GET_USERS query to validate the user on the BE